### ISSUE TITLE
Fatal error with 2021.01 friendica/src/Core/Renderer.php:91

### ISSUE THREAD

**Issue by StackKorora** [friendica/friendica#9999]
- [X] I have searched open and closed issues for duplicates

### Bug Description

I realize I might be a little fringe. Will detail as much as I can. For various reasons the server that I would like to run friendica on has to be CentOS 8 (fully updated 8.3) with Nginx+PHP-FPM. The server is also hosting other sites in this configuration successfully on subdomains.

Friendica Version: friendica-full-2021.01.tar.gz
OS: CentOS Linux release 8.3.2011
Nginx: nginx-1.14.1-9
PHP: 7.2.24
MariaDB: 10.3.27

This is my nginx config file based heavily on the friendica nginx example:
```
server {
  listen 80;
  server_name friendica.example.com;
  index index.php;
  root /var/www/html/friendica;
  rewrite ^ https://friendica.example.com$request_uri? permanent;
}
server {
  listen 443 ssl;
  server_name friendica.example.com;
  ssl on;
  ssl_certificate /etc/pki/tls/certs/ca.crt;
  ssl_certificate_key /etc/pki/tls/private/ca.key;
  ssl_session_timeout 5m;
  ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
  ssl_ciphers ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS;
  ssl_prefer_server_ciphers on;
  fastcgi_param HTTPS on;
  index index.php;
  charset utf-8;
  root /var/www/html/friendica;
  access_log /var/log/nginx/friendica.log;
  client_max_body_size 20m;
  client_body_buffer_size 128k;
  location / {
    try_files $uri /index.php?pagename=$uri&$args;
  }
  location ^~ /.well-known/ {
    allow all;
    rewrite ^ /index.php?pagename=$uri;
    root /var/www/letsencrypt;
  }
  include mime.types;
  location ~* \.php$ {
    try_files $uri =404;
    fastcgi_split_path_info ^(.+\.php)(/.+)$;
    fastcgi_pass php-handler;
    include fastcgi_params;
    fastcgi_index index.php;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    fastcgi_buffers 16 16k; 
    fastcgi_buffer_size 32k;
  }
  location ~* \.(tpl|md|tgz|log|out)$ {
    deny all;
  }
  location ~ /\. {
    deny all;
  }
  location ^~ /bin {
    deny all;
  }
}
```
I've loaded the DB per the instructions, but when I go to the web page for the install I only get a blank page. In the nginx error.log file I get this:

```
2021/03/06 21:36:59 [error] 49473#0: *732 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Friendica\Network\HTTPException\InternalServerErrorException: Friendica can't display this page at the moment, please contact the administrator. in /var/www/html/friendica/src/Core/Renderer.php:91
Stack trace:
#0 /var/www/html/friendica/src/Module/Special/HTTPException.php(118): Friendica\Core\Renderer::replaceMacros('file:/var/www/h...', Array)
#1 /var/www/html/friendica/src/App/Page.php(336): Friendica\Module\Special\HTTPException::content(Object(Friendica\Network\HTTPException\InternalServerErrorException))
#2 /var/www/html/friendica/src/App/Page.php(389): Friendica\App\Page->initContent(Object(Friendica\App\Module), Object(Friendica\App\Mode))
#3 /var/www/html/friendica/src/App.php(562): Friendica\App\Page->run(Object(Friendica\App), Object(Friendica\App\BaseURL), Object(Friendica\App\Mode), Object(Friendica\App\Module), Object(Friendica\Core\L10n), Object(Friendica\Util\Profiler), Object(Friendica\Core\Config\JitConfig), Object(Friendica\Core\PConfig\JitPConfig" while reading response header from upstream, client: 192.168.1.6, server: friendica.example.com, request: "GET / HTTP/2.0", upstream: "fastcgi://127.0.0.1:9000", host: "friendica.example.com"
```
I thought it might have something to do the the fastcgi rendering with php-fpm, but I cranked that logging all the way up in debug mode and it had nothing of note. Everything looks fine other wise. I can put data into a test.html file and access it. And everything that I've tested seems fine. But I'm at a complete loss for why friendica is giving me this error. Any thoughts?

**Comment by MrPetovan** [comment]
Thank you for the report, is your Friendica `view/smarty3` subfolder is writable by the user running PHP-FPM? It is required to compile templates and render pages.

**Comment by StackKorora** [comment]
Greetings @MrPetovan,
Yes, I did check that. I even set it to 777 for testing. Even putting SELinux into permissive didn't do the trick.
Thanks!

**Comment by MrPetovan** [comment]
Drat, we have a more detailed error message but both ways of displaying it depend on the node being installed already.

If you'd like to stick with us, we'll have to set up the configuration file manually to enable detailed logging.

**Comment by StackKorora** [comment]
Greetings @MrPetovan ,

So 99% certain it was a server configuration error. I'm just not entirely sure which one. I started by trying to crank up php-fpm and doing straces to figure out what the heck it was doing while also cranking up audit rules. Suddenly I got a page but it was very very angry with me. Loads of warnings and it was only displaying text jammed up on the left side.

I decided to start over.

Dumped the db and recreated it. Dumped everything in the friendica directory then re-extracted the tar.gz. I had SELinux in permissive but I reset the owner, permissions, and SELinux context on the friendica directory (as well as view/smarty3) and watched what happened.  And... it worked. :man_shrugging: 

I did see in the SELinux log that I needed to adjust it by running `setsebool -P httpd_unified 1` and after I re-enabled SELinux I was able to complete the installation without setting off anything else.

As of right now, it's working! :confetti_ball: 

Thank you so much for helping me out. I really don't know what went wrong with that first install but as it's my first time through who knows what I did tinkering with it that broke things. I wish I could give a better "here's what was wrong and what I did to fix it" report.
